docs/config/lua/config/local_echo_threshold_ms.md

```diff
@@ -0,0 +1,10 @@
+# `local_echo_threshold_ms`
+
+Specifies the maximum tolerated round-trip latency before using predictive
+echoing on remote terminals. The value is measured in milliseconds.
```

Comment:
Would it be the place to explain what `predictive echoing` means? Or put a link that explains it?
(I have no idea what it is)

Question for myself: if it's about predictions of remote terminal, why is the option named `local_echo_...`  ?

Reply:
Predictive local echo is a feature of remote multiplexer connections that is intended to hide latency.
The idea is that if we *predict* that a given keypress (say of the `a` key) might result in an `a` being printed at the cursor location, then we could locally show that before the remote host responds and updates the screen.  This can make a slow link feel fast if you are quickly typing prose, as the text appears to quickly update, but can also be jarring if the threshold is very close to the actual latency of the link--the cursor can appear to jiggle back and forth.

Reply:
I'm not sure what the best place would be, but for sure some more doc sounds like a good idea. Unless @wez already has something I can refer to I can try and write something down here.

To answer your question in the meantime, the predictive echoing feature is basically a nice trick to try and hide latency when using e.g. an ssh domain where the round-trip latency can be high. Other tools (mosh for instance) have a similar feature. The idea is for the client to make a prediction about the server's response to key events and immediately echo that locally. And when the server's response arrives later on, if the client's speculation was correct it's all good, otherwise the client might need to back off what it echoed locally, which can lead to visible 'noise'.